Sources/MongoSwift/ConnectionString.swift

```diff
@@ -40,15 +40,28 @@ internal class ConnectionString {
             }
         }
 
+        if let heartbeatFrequencyMS = options?.heartbeatFrequencyMS {
+            guard let value = Int32(exactly: heartbeatFrequencyMS), value >= 500 else {
+                throw MongoError.InvalidArgumentError(
+                    message: "heartbeatFrequencyMS must be between 500 and \(Int32.max)"
+                )
+            }
+
+            guard mongoc_uri_set_option_as_int32(self._uri, MONGOC_URI_HEARTBEATFREQUENCYMS, value) else {
+                throw MongoError.InvalidArgumentError(
+                    message: "Failed to set heartbeatFrequencyMS to \(value)"
+                )
+            }
+        }
+
         if let maxPoolSize = options?.maxPoolSize {
-            // libmongoc casts the value to a uint32_t
-            guard maxPoolSize > 0 && maxPoolSize <= UInt32.max else {
+            guard let value = Int32(exactly: maxPoolSize), value > 0 else {
```

Comment:
your comment made me realize I could apply the same logic here to do the cast/error check at once. that made me realize we had a bug here (introduced by me): `UInt32.max` is larger than `Int32.max`. if you provided a value that was larger that could not fit into an `Int32` but could fit into a `UInt32`, we wouldn't error here, but we would then crash doing the cast below.
so while libmongoc stores the value as a `uInt32_t`, the API restricts the max value to `Int32.max`.

Reply:
nice!